src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -445,6 +524,22 @@ private JsonRpcError ReadError(JToken json)
             };
         }
 
+        private void ClearProgressObject(JToken requestId)
+        {
+            long reqId = requestId.Value<long>();
```

Comment:
please add `Requires.NotNull(requestId, nameof(requestId));` here, since you dereference it.